### PR TITLE
gh-95504: Fix negative numbers in PyUnicode_FromFormat

### PR DESCRIPTION
**Pull request by encukou**
This is the non-refactoring part of #95505.

(Plus the ACKS fixup I forgot to include in #95819 😅)

* Issue: gh-95504

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -475,6 +475,8 @@ Porting to Python 3.12
   copied as-is to the result string, and any extra arguments discarded.
   (Contributed by Serhiy Storchaka in :gh:`95781`.)
 
+* Fixed wrong sign placement in ``PyUnicode_FromFormat``.
```
Comment (philg314) [inline Doc/whatsnew/3.12.rst]:
```suggestion
* Fixed wrong sign placement in :c:func:`PyUnicode_FromFormat` and 
  :c:func:`PyUnicode_FromFormatV`.
```

File: Misc/NEWS.d/next/C API/2022-07-31-21-58-27.gh-issue-95504.wy7B1F.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fix sign placement when specifying width or precision in
+``PyUnicode_FromFormat``. Patch by Philip Georgi.
```
Comment (philg314) [inline Misc/NEWS.d/next/C API/2022-07-31-21-58-27.gh-issue-95504.wy7B1F.rst]:
```suggestion
:c:func:`PyUnicode_FromFormat` and :c:func:`PyUnicode_FromFormatV`.
Patch by Philip Georgi.
```

**Review by philg314 — APPROVED** [review]
There are some minor adjustment to the news entries based on the news entries for the `%%` changes but other than that everything is in order. Thank you!

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -475,7 +475,7 @@ Porting to Python 3.12
   copied as-is to the result string, and any extra arguments discarded.
   (Contributed by Serhiy Storchaka in :gh:`95781`.)
 
-* Fixed wrong sign placement in :c:func:`PyUnicode_FromFormat` and
```
Comment (philg314) [inline Doc/whatsnew/3.12.rst]: Sorry!

Comment (encukou) [reply Doc/whatsnew/3.12.rst]: Oh, no problem!  I'm just not a fan of the nitpycky tests :)

**Pull request merged**